src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 8;
+    private T[] storage = (T[])new Object[DEFAULT_CAPACITY];
+    private int size;
+
     @Override
     public void add(T value) {
-
+        if (size == storage.length) {
+            changeLength(size + 1);
+        }
+        storage[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= size) {
+            if (size == storage.length) {
+                changeLength(size + 1);
+            }
+            for (int i = size; i > index; i--) {
+                storage[i] = storage[i - 1];
+            }
+            storage[index] = value;
+            size++;
+            return;
+        }
+        throw new ArrayListIndexOutOfBoundsException("Wrong index");
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list.size() + size > storage.length) {
+            changeLength(list.size() + size + 1);
+        }
+        int newSize = list.size() + size;
+        for (int i = size, j = 0; i < newSize; i++, j++) {
+            storage[i] = list.get(j);
+        }
+        size = newSize;
     }
 
     @Override
-    public T get(int index) {
-        return null;
+    public T remove(int index) {
+        T tmp;
+        if (index >= 0 && index < size) {
+            tmp = storage[index];
+            for (int i = index; i < size - 1; i++) {
+                storage[i] = storage[i + 1];
+            }
```

Comment:
[Common mistakes](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html)
Use [System.arraycopy()](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#arraycopy-java.lang.Object-int-java.lang.Object-int-int-) to move your array elements.